config/webserver/emmental_flask/db.py

```diff
@@ -0,0 +1,98 @@
+import mysql.connector, os
+import base64
+
+# get reference to connected DB
+def get_db():
+    return connect_db()
+# connect to db
+def connect_db():
+  dbconnection = mysql.connector.connect(host=os.environ['MARIADB_IP'], user='databaseuser', password='securedatabasepassword', database='emmental')
+  if dbconnection:
+    print("SUCCESS")
+  if not dbconnection:
+    print("FAILURE")
+  return dbconnection
+
+# close db connection
+def close_db(dbconnection):
+  dbconnection.close()
+
+#make a 'select' type query - must not be destructive. Does not commit to db.
+def query(sqlquery):
+  dbconnection = get_db()
+  cursor = dbconnection.cursor()
+  cursor.execute("use emmental;")
+  cursor.execute(sqlquery)
+  close_db(dbconnection)
+  for result in cursor:
+    print(result)
+
+# insert/add user into mariadb
+def add_user(username, password):
+  dbconnection = get_db()
+  userID = generate_userID()
+  cursor = dbconnection.cursor()
+  dbusername = username
+  cursor.execute("INSERT INTO Users (userID, username, password) VALUES (%s,%s,%s)", (userID, dbusername, password))
```

Comment:
Change the %s for the userID to UUID_TO_BIN(%s)

Reply:
Converting to binary makes the fields larger than we really need to store. It's easier to just store the full 36 char UUID instead.